### DISCUSSION TITLE
Is there a way to detect preload on the server?

### DISCUSSION THREAD

**Discussion by LaundroMat** [unpoly/unpoly#689] [Q&A] [answered]
I'm using Django 5.1 and the python-unpoly library. There doesn't seem to be a way of detecting whether a request is an Unpoly preload? I'd need to know because I store the pages a user has visited, but preloaded pages should not be tallied ofcourse (unless actually visited by the user).

**Comment by triskweline** [comment, marked as answer]
While you could add a custom header on the `up:link:preload` event, I'm not sure this would be a good solution in your case. After all, when the preloaded content does get used, you do not get another request as long as the cache is still fresh.

Maybe consider tracking the page my making another request on the client, after the fragment was rendered. See [Tracking page views](https://unpoly.com/analytics).
